src/extensions/agent_modules/cloud_communication/eg_mqtt_broker_client/src/adu_enrollment_management.c

```diff
@@ -18,143 +18,259 @@
+ * @param correlationId The correlation id to set.
  */
-void Free_Enrollment_Data(ADU_ENROLLMENT_DATA* data)
+static void
+EnrollmentData_SetCorrelationId(ADUC_Enrollment_Request_Operation_Data* enrollmentData, const char* correlationId)
 {
-    if (data == NULL)
+    if (enrollmentData == NULL || correlationId == NULL)
     {
+        Log_Error("Null input (enrollmentData:%p, correlationId:%p)", enrollmentData, correlationId);
         return;
     }
 
-    free(data->enr_resp_duInstance);
-    free(data->enr_resp_resultDescription);
-    free(data);
-    memset(&data, 0, sizeof(data));
+    strncpy(
+        enrollmentData->enrReqMessageContext.correlationId,
+        correlationId,
+        sizeof(enrollmentData->enrReqMessageContext.correlationId) - 1);
+    enrollmentData->enrReqMessageContext
+        .correlationId[sizeof(enrollmentData->enrReqMessageContext.correlationId) - 1] = '\0';
+}
+
+static int json_print_properties(const mosquitto_property* properties)
+{
+    int identifier;
+    uint8_t i8value = 0;
+    uint16_t i16value = 0;
+    uint32_t i32value = 0;
+    char *strname = NULL, *strvalue = NULL;
+    char* binvalue = NULL;
+    const mosquitto_property* prop = NULL;
+
+    for (prop = properties; prop != NULL; prop = mosquitto_property_next(prop))
+    {
+        identifier = mosquitto_property_identifier(prop);
+        switch (identifier)
+        {
+        case MQTT_PROP_PAYLOAD_FORMAT_INDICATOR:
+            mosquitto_property_read_byte(prop, MQTT_PROP_PAYLOAD_FORMAT_INDICATOR, &i8value, false);
+            break;
+
+        case MQTT_PROP_MESSAGE_EXPIRY_INTERVAL:
+            mosquitto_property_read_int32(prop, MQTT_PROP_MESSAGE_EXPIRY_INTERVAL, &i32value, false);
+            break;
+
+        case MQTT_PROP_CONTENT_TYPE:
+        case MQTT_PROP_RESPONSE_TOPIC:
+            mosquitto_property_read_string(prop, identifier, &strvalue, false);
+            if (strvalue == NULL)
+                return MOSQ_ERR_NOMEM;
+            free(strvalue);
+            strvalue = NULL;
+            break;
+
+        case MQTT_PROP_CORRELATION_DATA:
+            mosquitto_property_read_binary(prop, MQTT_PROP_CORRELATION_DATA, (void**)&binvalue, &i16value, false);
+            if (binvalue == NULL)
+                return MOSQ_ERR_NOMEM;
+            free(binvalue);
+            binvalue = NULL;
+            break;
+
+        case MQTT_PROP_SUBSCRIPTION_IDENTIFIER:
+            mosquitto_property_read_varint(prop, MQTT_PROP_SUBSCRIPTION_IDENTIFIER, &i32value, false);
+            break;
+
+        case MQTT_PROP_TOPIC_ALIAS:
+            mosquitto_property_read_int16(prop, MQTT_PROP_TOPIC_ALIAS, &i16value, false);
+            break;
+
+        case MQTT_PROP_USER_PROPERTY:
+            mosquitto_property_read_string_pair(prop, MQTT_PROP_USER_PROPERTY, &strname, &strvalue, false);
+            if (strname == NULL || strvalue == NULL)
+                return MOSQ_ERR_NOMEM;
+
+            free(strvalue);
```

Comment:
This makes no sense to me.  why is it free+null these immediately after mosquitto_property_read_string_pair allocate and output to them?